dataset/src/test/scala/frameless/functions/NonAggregateFunctionsTests.scala

```diff
@@ -0,0 +1,450 @@
+package frameless
+package functions
+
+import frameless.functions.nonAggregate._
+import org.apache.spark.sql.Encoder
+import org.scalacheck.Gen
+import org.scalacheck.Prop._
+
+class NonAggregateFunctionsTests extends TypedDatasetSuite {
+
+  test("abs") {
+    val spark = session
+    import spark.implicits._
+
+    def prop[A: CatalystNumeric : TypedEncoder : Encoder](value: List[A])= {
+      val cDS = session.createDataset(value)
+      val resCompare = cDS
+        .select(org.apache.spark.sql.functions.abs(cDS("value")))
+        .map(_.getAs[A](0))
+        .collect().toList
+
+
+      val typedDS = TypedDataset.create(value.map(X1(_)))
+      val res = typedDS
+        .select(abs(typedDS('a)))
+        .collect()
+        .run()
+        .toList
+
+      res ?= resCompare
+    }
+
+
+    check(forAll(prop[Int] _))
+    check(forAll(prop[Long] _))
+    check(forAll(prop[Short] _))
+    //check(forAll(prop[BigDecimal] _)) untyped abs implementation will convert this to java.math.BigDecimal which results in class cast exceptions.
```

Comment:
Then it should not type check! For inspiration you might have a look at this commit where I fixed this exact problem for BigDecimal division: https://github.com/typelevel/frameless/pull/179/commits/6f34f333e564f6da3035aa0376e38eab48245208

Reply:
fixed by adding a typeclass like divisible